File: frontend/src/types/history.ts
// Mirrors backend/app/api/schemas/history.py

import type { Citation } from './chat'
import type { ConfidenceLevel, MessageRole } from './enums'

export interface HistoryTurn {
  role: MessageRole
  content: string
  citations: Citation[]
  confidence: ConfidenceLevel | null
  created_at: string
}

export interface HistoryResponse {
  session_id: string
  title: string | null
  turns: HistoryTurn[]
}

export interface HistoryDeleteResponse {
  sessions_cleared: number
  messages_deleted: number
}

File: backend/app/api/schemas/history.py
"""Schemas for GET/DELETE /api/v1/history (SDD Section 10.5, 10.6)."""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.api.schemas.chat import Citation
from app.core.constants import ConfidenceLevel, MessageRole


class HistoryTurn(BaseModel):
    role: MessageRole
    content: str
    citations: list[Citation]
    confidence: ConfidenceLevel | None
    created_at: datetime


class HistoryResponse(BaseModel):
    session_id: uuid.UUID
    title: str | None
    turns: list[HistoryTurn]


class HistoryDeleteResponse(BaseModel):
    sessions_cleared: int
    messages_deleted: int
